src/app/store/transactions/transaction.hooks.ts

```diff
@@ -188,20 +190,18 @@ export function useSoftwareWalletTransactionBroadcast() {
             attachment,
             networkUrl: network.url,
           });
-          finalizeTxSignature(requestToken, result);
-          if (typeof result.txId === 'string') {
-            set(currentAccountLocallySubmittedTxsState, {
-              [result.txId]: {
-                rawTx: result.txRaw,
-                timestamp: todaysIsoDate(),
-              },
+          if (typeof result.txId === 'string' && !accountTxIds.includes(result.txId)) {
```

Comment:
To your other comment on using an object, this check wouldn't be necessary. Using an object prevents duplicates as you can only ever have one value assigned to a given key.

It's not the end of the world to use an array, if you like, but there are merits to using maps/dictionaries. When using createEntityAdapter, it'll be an object. It's a common approach to take when ensuring state is normalised, see https://redux.js.org/usage/structuring-reducers/normalizing-state-shape

Reply:
Sure, sorry it is indeed unrelated to object discussion. 

Though, not sure why it's needed? 🤔  In what circumstance would a pending tx appear before it's submitted?

Reply:
Not quite sure, but it was a check used in the send form so added here. Maybe just a safety check to make sure it hasn't already become pending. Maybe it could happen that fast?

Reply:
I see. I commend your anti-race condition approach. I don't think there's any chance of this happening, but I suppose it doesn't hurt.

Reply:
Haha, I can remove it and see if we run into any issues during testing.